### PR TITLE
Custom callout improvements

### PR DESCRIPTION
**Pull request by ukrbublik**
Added new feature - handling press on callout subview for iOS.
This required adding new component `<CalloutSubview />` with `onPress` event.
See "Custom Callouts" (Callouts.js) example.

Added prop `alphaHitTest` for `Callout`.
If `true`, press on transparent areas in custom callout will be passed to map (will hide current callout or open callout of another marker positioned at touch point).
It's very useful for custom callouts with big transparent areas. At example bottom part of custom callout is transparent (except arrow).

Added prop `zoomTapEnabled` for `MapView`.
If `false` user won't be able to double tap to zoom the map. BUT it will greatly decrease delay of single tap gesture recognition in Google Maps for iOS so callouts will open much faster!

Added method `redrawCallout` for `Marker`. Useful for Google Maps on iOS.

Also added method `getMarkersFrames(onlyVisible=false)` for `MapView`.
It gathers all (or only visible) markers and return their frames and centers in pixels.
See "FitToCoordinates" example.
It's not related to callouts but uses same new methods I added in this PR and can be useful on JS-side.

### Does any other open PR do the same thing?

It's improved version of my past PR #2538 

### What issue is this PR fixing?

#2279

### How did you test this PR?

Change affects iOS, both Google & Apple map providers.
Tested in iOS simulator and real devices.

### CODE REVIEW

**Comment by christopherdro** [comment]
@ukrbublik Should this replace #2538 or do both go together?

**Comment by ukrbublik** [comment]
> @ukrbublik Should this replace #2538 or do both go together?
Yes, it fully replaces #2538

**Comment by christopherdro** [comment]
@ukrbublik Thanks for the update. Any chance you can resolve the merge conflicts in this PR for proper review?

File: lib/ios/AirGoogleMaps/AIRGoogleMap.m [hunk 1]
```diff
@@ -528,6 +545,203 @@ + (GMSCameraPosition*) makeGMSCameraPositionFromMap:(GMSMapView *)map andMKCoord
...
+            [gestureRecognizer removeTarget:target action:action];
+        }
+        [gestureRecognizer addTarget:self action:@selector(ourMapGestureHandler:)];
+        
+        [_origGestureRecognizersMeta setObject:@{@"targets": origTargetsActions}
+                                        forKey:grHash];
+    }
+}
+
+- (id)ourMapGestureHandler:(UIGestureRecognizer*)gestureRecognizer {
```
Comment (rborn) [inline lib/ios/AirGoogleMaps/AIRGoogleMap.m]: Maybe find a better name for this method, unless it follows something already in?

File: lib/ios/AirMaps/AIRMapMarker.m [hunk 2]
```diff
@@ -205,13 +205,52 @@ - (void)_handleTap:(UITapGestureRecognizer *)recognizer {
...
+//            AIRMapMarker* markerAtTapPoint = [self.map markerAtPoint:touchPointMap];
+//            if (markerAtTapPoint != nil) {
+//                [markerAtTapPoint showCalloutView];
+//            } else {
+//                [self.map.calloutView dismissCalloutAnimated:YES];
+//            }
+//            return;
+//        }
+        
+        UIView *calloutMaybe = [marker.map.calloutView hitTest:touchPoint withEvent:nil];
```
Comment (rborn) [inline lib/ios/AirMaps/AIRMapMarker.m]: can we find a better name for `calloutMaybe`

File: package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
     "start:android": "adb shell am start -n com.airbnb.android.react.maps.example/.MainActivity",
     "run:android": "./gradlew installDebug && npm run start:android",
     "lint": "eslint ./",
+    "format": "eslint ./ --fix",
```
Comment (rborn) [inline package.json]: do we want this in this commit/PR ?

**Comment by ukrbublik** [comment]
@rborn Thanks for review. I will fix/reply soon.
As for Android, sadly I don't have skills in Android development.

File: docs/callout.md [hunk 4]
```diff
@@ -5,10 +5,36 @@
...
+| `alphaHitTest` | `Boolean` | `false` | If `true`, clicks on transparent areas in callout will be passed to map. **Note**: iOS only.
 
 ## Events
 
 | Event Name | Returns | Notes
 |---|---|---|
 | `onPress` |  | Callback that is called when the user presses on the callout
+
+
+&NewLine;
```
[the remark this answers is not in the document]
Comment (ukrbublik) [reply docs/callout.md]: Should be line breaks, to make bigger distance between "<Callout /> Component API" and "<CalloutSubview /> Component API"
However seems like GitHub's markdown ignore them, so I'll remove

File: lib/ios/AirGoogleMaps/AIRGoogleMap.m (hunk 1 above)
Comment (ukrbublik) [reply lib/ios/AirGoogleMaps/AIRGoogleMap.m]: Renamed to extendedMapGestureHandler

File: lib/ios/AirMaps/AIRMapMarker.m (hunk 2 above)
Comment (ukrbublik) [reply lib/ios/AirMaps/AIRMapMarker.m]: Sure, renamed to `calloutView`

File: package.json (hunk 3 above)
Comment (ukrbublik) [reply package.json]: No

**Comment by WeslleyNasRocha** [comment]
i cant make any promises, but i will try to implement the `getMarkersFrames()` on android, cus im in need of it

**Comment by christopherdro** [comment]
@ukrbublik There seems to be a memory leak coming from https://github.com/react-native-community/react-native-maps/pull/2581/files#diff-c9a846285e908a781d4c99257a3f17fbR596

Any ideas?

**Comment by christopherdro** [comment]
@alvelig Any ideas?

**Comment by ukrbublik** [comment]
@christopherdro
`free(ivars)` is missing

**Comment by christopherdro** [comment]
@ukrbublik Thanks that did it.

**Comment by ukrbublik** [comment]
Seems like from #2782 leek is still present, I will take a look today

**Pull request merged**